src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -268,12 +263,8 @@ private FedoraResource getBaseVersion(final FedoraResource resource) {
      *  to whether the given acl:accessToClass values contain any of the rdf:type values provided
      *  when creating the predicate.
      */
-    private final Function<List<String>, Predicate<WebACAuthorization>> accessToClass = uris -> x -> {
-        return uris.stream()
-                   .distinct()
-                   .filter(y -> x.getAccessToClassURIs().contains(y))
-                   .findFirst()
-                   .isPresent();
+    private final Function<List<String>, Predicate<WebACAuthorization>> accessToClass = uris -> auth -> {
```

Comment:
After our recent discussion about code style, you can drop the brackets and `return` here. I would find that a lot easier to read.